src/udev/udevadm-trigger.c

```diff
@@ -48,6 +49,19 @@ static int exec_list(
                         continue;
                 }
 
+                if (arg_database_match >= 0) {
+                        r = sd_device_get_is_initialized(d);
+
+                        if (r > 0) {
+                                if (!arg_database_match)
+                                        continue;
+                        } else if (!r) {
+                                if (arg_database_match)
+                                        continue;
+                        } else
+                                log_warning_errno(r, "Failed to check presence of %s in udev db", syspath);
```

Comment:
if this fails, ignore the device please. i.e. add a `continue` here.


I'd really write this this way btw:

```c
if (arg_database_matc >= 0) {
        r = sd_device_get_initialized(d);
        if (r < 0) {
                log_warning_errno(r, "Failed to check presence of %s in udev database, ignoring device: %m", syspath);
                continue;
        }

        if (!!r != !!arg_database_match)
                continue;
}
```